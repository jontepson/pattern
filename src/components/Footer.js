import React from 'react'
import { StyleSheet } from 'react-native'
import Paragraph from './Paragraph'

export default function Header(props) {
  return <Paragraph style={styles.footer}>
  By Svenska Elsparkcyklar AB
  </Paragraph>
}

const styles = StyleSheet.create({
  footer: {
    position: 'absolute', left: 0, right: 0, bottom: 10, textAlign: 'center'
  }
})