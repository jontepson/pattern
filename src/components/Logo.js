import React from 'react'
import { StyleSheet } from 'react-native'
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome'
import { faAtlassian } from '@fortawesome/free-brands-svg-icons'
import { theme } from '../core/theme';
import Header from './Header';
import Paragraph from './Paragraph';
export default function Logo() {

  return (
    <>
  <FontAwesomeIcon style={styles.logo} icon={ faAtlassian } size={ 100 } />
  <Header>Ajjo</Header>
  <Paragraph>
        No more walking
  </Paragraph>
  </>
  )
}

const styles = StyleSheet.create({
  logo: {
    color: theme.colors.primary
  },
})