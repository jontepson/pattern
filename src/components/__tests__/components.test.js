import React from 'react';
import renderer, { act } from 'react-test-renderer';

import Background from '../Background';
import BackButton from '../BackButton';
import Button from '../Button';
import Footer from '../Footer';
import Header from '../Header';
import Logo from '../Logo';
import Paragraph from '../Paragraph';


describe('<Components />', () => {
    it('<Background /> renders correctly', async () => {
        let tree;
        act(() => {
            tree = renderer.create(<Background />).toJSON();
        })
        expect(tree).toMatchSnapshot();
    });

    it('<BackButton /> renders correctly', async () => {
        let tree;
        act(() => {
            tree = renderer.create(<BackButton />).toJSON();
        })
        expect(tree).toMatchSnapshot();
    });

    it('<Button /> renders correctly', async () => {
        let tree;
        act(() => {
            tree = renderer.create(<Button />).toJSON();
        })
        expect(tree).toMatchSnapshot();
    });

    it('<Footer /> renders correctly', async () => {
        let tree;
        act(() => {
            tree = renderer.create(<Footer />).toJSON();
        })
        expect(tree).toMatchSnapshot();
    });

    it('<Header /> renders correctly', async () => {
        let tree;
        act(() => {
            tree = renderer.create(<Header />).toJSON();
        })
        expect(tree).toMatchSnapshot();
    });

    it('<Logo /> renders correctly', async () => {
        let tree;
        act(() => {
            tree = renderer.create(<Logo />).toJSON();
        })
        expect(tree).toMatchSnapshot();
    });

    it('<Paragraph /> renders correctly', async () => {
        let tree;
        act(() => {
            tree = renderer.create(<Paragraph />).toJSON();
        })
        expect(tree).toMatchSnapshot();
    });
  });