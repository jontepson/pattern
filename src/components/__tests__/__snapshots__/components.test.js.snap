// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Components /> <BackButton /> renders correctly 1`] = `null`;

exports[`<Components /> <Background /> renders correctly 1`] = `null`;

exports[`<Components /> <Button /> renders correctly 1`] = `null`;

exports[`<Components /> <Footer /> renders correctly 1`] = `null`;

exports[`<Components /> <Header /> renders correctly 1`] = `null`;

exports[`<Components /> <Logo /> renders correctly 1`] = `null`;

exports[`<Components /> <Paragraph /> renders correctly 1`] = `null`;
